### PR TITLE
Updated Newtonsoft to 13.0.1

### PR DESCRIPTION
**Pull request by brentschmaltz**
updated snapshot of newtonsoft to 13.0.1

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/opensource/json/ConstructorHandling.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ namespace Microsoft.IdentityModel.Json
     /// <summary>
     /// Specifies how constructors are used when initializing objects during deserialization by the <see cref="JsonSerializer"/>.
     /// </summary>
-    internal enum ConstructorHandling
+    public enum ConstructorHandling
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/ConstructorHandling.cs:31]: >public [](http://example.com/codeflow?start=4&length=6)

These need to be public now?

File: src/Microsoft.IdentityModel.Tokens/opensource/json/DefaultJsonNameTable.cs [hunk 2]
```diff
@@ -77,7 +81,12 @@ public override string Get(char[] key, int start, int length)
             hashCode -= hashCode >> 5;
 
             // make sure index is evaluated before accessing _entries, otherwise potential race condition causing IndexOutOfRangeException
-            var index = hashCode & _mask;
+#if NET20 || NET35 || NET40
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/DefaultJsonNameTable.cs:84]: >NET35 || NET40 [](http://example.com/codeflow?start=13&length=14)

do we need net35 and net40?

File: src/Microsoft.IdentityModel.Tokens/opensource/json/JsonReader.Async.cs [hunk 3]
```diff
@@ -33,8 +33,27 @@
 
 namespace Microsoft.IdentityModel.Json
 {
-    internal abstract partial class JsonReader
+#nullable enable
+    public abstract partial class JsonReader
+#if HAVE_ASYNC_DISPOABLE
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/JsonReader.Async.cs:38]: >HAVE_ASYNC_DISPOABLE [](http://example.com/codeflow?start=4&length=20)

nit: spelling - DISPOSABLE - missing an "s"

File: src/Microsoft.IdentityModel.Tokens/opensource/json/JsonValidatingReader.cs [hunk 4]
```diff
@@ -41,6 +41,8 @@
 
 #endif
 
+#nullable disable
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/JsonValidatingReader.cs:44]: >disable [](http://example.com/codeflow?start=10&length=7)

Is this needed?

File: src/Microsoft.IdentityModel.Tokens/opensource/json/JsonWriter.Async.cs [hunk 5]
```diff
@@ -291,7 +305,7 @@ internal Task InternalWriteEndAsync(JsonContainerType type, CancellationToken ca
                 if (_currentState == State.Property)
                 {
                     t = WriteNullAsync(cancellationToken);
-                    if (!t.IsCompletedSucessfully())
+                    if (!t.IsCompletedSuccessfully())
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/JsonWriter.Async.cs:308]: took me awhile to figure out the dif on this one!

File: src/Microsoft.IdentityModel.Tokens/opensource/json/SerializationBinder.cs [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ﻿
-#if (DOTNET || PORTABLE40 || PORTABLE || NET_CORE)
+#if (DOTNET || PORTABLE40 || PORTABLE)
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Tokens/opensource/json/SerializationBinder.cs:2]: >PORTABLE40 || PORTABLE [](http://example.com/codeflow?start=15&length=22)

do we need these?

**Comment by brentschmaltz** [comment]
Have to mark the publics to internal....

File: src/Microsoft.IdentityModel.Tokens/opensource/json/ConstructorHandling.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/opensource/json/ConstructorHandling.cs:31]: My bad, need to make these internal.
Will reissue the PR.

**Pull request closed without merging**